### PR TITLE
Adding private registry support for yarn berry

### PR DESCRIPTION
**Pull request by pavera**
Initial support for private registries for yarn berry.  The credential injection code needs additional work specifically around detecting if credentials are needed for a given package update. I will also investigate writing a .yarnrc.yml builder analogous to the npmrc builder we currently have.

### CODE REVIEW

**Comment by jurre** [comment]
>  The credential injection code needs additional work 

@pavera you wanna do that in this PR or in a follow-up?

**Comment by pavera** [comment]
> @pavera you wanna do that in this PR or in a follow-up?
It's a bit tied up with the yarnrc.yml builder so I think I'll endeavor to do both of those in a subsequent PR.

File: npm_and_yarn/spec/fixtures/npm_responses/scoped_private_details.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "publishConfig" : {
+      "@dsp-testing:registry" : "https://pavera.jfrog.io/artifactory/api/npm/pavera-npm/"
+    },
+    "_id" : "@dsp-testing/my-package@1.0.4",
+    "dist" : {
+      "tarball" : "https://pavera.jfrog.io/artifactory/api/npm/pavera-npm/@dsp-testing/my-package/-/@dsp-testing/my-package-1.0.4.tgz",
+      "shasum" : "df5cfe11999e1691e97a4883c2cf50615cc36ff5",
+      "integrity" : "sha512-RrD1S1yqIoj3VGoej0PqHa4ZmVks4984yILeqC01fOTV/wHcsgtRGMDCrlrNtba5pGEjaXHl4OZtKNRlg4FoTg=="
+    }
+  }
```
Comment (jeffwidman) [inline npm_and_yarn/spec/fixtures/npm_responses/scoped_private_details.json:21]: nit: missing newline

I noticed this a few other places in this PR... are these files auto-generated by yarn or or is there some reason you prefer not to have them? Or have you just not gotten around to configuring VSCode to auto-append new lines? https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

**Review by jeffwidman — COMMENTED** [review]
Do we need these 400+ KB zip files for these particular set of tests?

See discussion here: https://github.com/dependabot/dependabot-core/pull/5706#pullrequestreview-1106194318

Since we can't add them to `.gitignore`, it's just something you'll have to remember to doublecheck... maybe here is one of those places you actually need it?

**Comment by jeffwidman** [comment]
Will this address https://github.com/dependabot/dependabot-core/issues/4669 as a side-effect? 

I'm not familiar enough with Yarn / NPM ecosystem to know...

File: composer/helpers/.php_cs.cache [hunk 2]
```diff
@@ -0,0 +1 @@
..._comma_in_array":true,"blank_line_after_namespace":true,"constant_case":true,"elseif":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_import_per_statement":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"visibility_required":true,"encoding":true,"full_opening_tag":true,"declare_strict_types":true,"is_null":{"use_yoda_style":false},"list_syntax":{"syntax":"short"},"modernize_types_casting":true,"no_multiline_whitespace_before_semicolons":true,"no_useless_else":true,"no_useless_return":true,"php_unit_construct":true,"php_unit_dedicate_assert":true,"phpdoc_order":true,"ternary_to_null_coalescing":true,"void_return":true},"hashes":{"src\/UpdateChecker.php":37664165,"src\/ExceptionIO.php":1817177428,"src\/Updater.php":1639505835,"src\/DependabotPluginManager.php":3091995801,"src\/DependabotInstallationManager.php":3086233364,"src\/Hasher.php":2787072973}}
```
[the remark this answers is not in the document]
Comment (jurre) [reply composer/helpers/.php_cs.cache:1]: It was accidentally added in the parent PR, but I've removed it there so after a rebase we should be ok again.

File: composer/helpers/v2/.php_cs.cache [hunk 3]
```diff
@@ -0,0 +1 @@
...comma_in_array":true,"blank_line_after_namespace":true,"constant_case":true,"elseif":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_import_per_statement":true,"single_line_after_imports":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"visibility_required":true,"encoding":true,"full_opening_tag":true,"declare_strict_types":true,"is_null":{"use_yoda_style":false},"list_syntax":{"syntax":"short"},"modernize_types_casting":true,"no_multiline_whitespace_before_semicolons":true,"no_useless_else":true,"no_useless_return":true,"php_unit_construct":true,"php_unit_dedicate_assert":true,"phpdoc_order":true,"ternary_to_null_coalescing":true,"void_return":true},"hashes":{"src\/UpdateChecker.php":3743743534,"src\/ExceptionIO.php":2517693200,"src\/Updater.php":3184456205,"src\/DependabotPluginManager.php":3180655585,"src\/DependabotInstallationManager.php":480511861,"src\/Hasher.php":1523553822}}
```
[the remark this answers is not in the document]
Comment (jurre) [reply composer/helpers/v2/.php_cs.cache:1]: It was accidentally added in the parent PR, but I've removed it there so after a rebase we should be ok again.

**Comment by jurre** [comment]
> Will this address https://github.com/dependabot/dependabot-core/issues/4669 as a side-effect?

I don't think so, but it does seem like the situation is better for yarn berry

**Comment by pavera** [comment]
We're going to revisit this after the initial merge of yarn_berry support.  Closing this PR

**Pull request closed without merging**